src/LidoDAO.ts

```diff
@@ -22,12 +23,24 @@ export function handleSetApp(event: SetAppEvent): void {
       // updating only in case of a new contract address
       if (entity.impl != event.params.app) {
         const repo = AppRepo.bind(Address.fromString(repoAddr))
-        const latest = repo.getLatestForContractAddress(event.params.app)
-        const semVer = latest.getSemanticVersion()
 
-        entity.major = semVer[0]
-        entity.minor = semVer[1]
-        entity.patch = semVer[2]
+        const triedLatest = repo.try_getLatestForContractAddress(event.params.app)
+        if (triedLatest.reverted) {
+          if (network !== 'holesky') {
```

Comment:
I suggest changing the check logic in case of error
- If the network is not Mainnet, then set version 0
- Otherwise throw an error

The main difference is not to check only one test network - Holesky.